### PR TITLE
adds blank line ignore

### PR DESCRIPTION
**Pull request by lattice0**
ignores blank line for cases where sdp has a blank line and adds test example from a real camera that has this 'problem'

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -734,15 +738,14 @@ impl<'iter, 'item, I: Iterator<Item = (usize, &'item [u8])>> FallibleIterator
                 })??;
 
             if key == self.current {
```
Comment (sdroege) [inline src/parser.rs]: You could also do

```suggestion
            return if key == self.current {
```

And add a `;` at the final `}`. But works for me either way :)

File: src/parser.rs [hunk 2]
```diff
@@ -715,7 +715,11 @@ impl<'iter, 'item, I: Iterator<Item = (usize, &'item [u8])>> FallibleIterator
     type Error = ParserError;
 
     fn next(&mut self) -> Result<Option<Self::Item>, Self::Error> {
-        if let Some((n, line)) = self.it.peek() {
+        while let Some((n, line)) = self.it.peek() {
+            if line.len() == 0 {
```
Comment (sdroege) [inline src/parser.rs]:
```suggestion
            if line.is_empty() {
```

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise :) Can you squash all commits into a single one with a meaningful commit message and `git push --force` them into this PR?

Also feel free to add a second commit with the `Cargo.toml` changes to reduce the number of dependencies from `bstr`.

File: src/parser.rs [hunk 3]
```diff
@@ -737,12 +737,12 @@ impl<'iter, 'item, I: Iterator<Item = (usize, &'item [u8])>> FallibleIterator
...
-                return Ok(self.it.next().map(|(n, line)| (n, &line[2..])))
+            return if key == self.current {
+                Ok(self.it.next().map(|(n, line)| (n, &line[2..])))
             } else if self.expected_next.contains(&key) {
-                return Ok(None)
+                Ok(None)
             } else {
-                return Err(ParserError::UnexpectedLine(*n, key))
+                Err(ParserError::UnexpectedLine(*n, key))
             }
```
Comment (sdroege) [inline src/parser.rs]:
```suggestion
            };
```

**Comment by lattice0** [comment]
Did you like the new commit?

**Comment by sdroege** [comment]
> Did you like the new commit?

Yes, apart from the missing `;` and the missing `cargo fmt` run this is good to be merged now :)

**Comment by lattice0** [comment]
I think it's ok now

**Review by sdroege — COMMENTED** [review]
Looks good to me (I'll fix the clippy warnings) but can you write a more descriptive commit message? Something like

```
Ignore blank lines in the SDP

While the RFC does not specify handling of blank lines, various software inserts them and skipping over them would make it possible to parse their SDPs.
```

**Comment by lattice0** [comment]
What do you think of it now?

**Review by sdroege — APPROVED** [review]

**Comment by sdroege** [comment]
Thanks, looks all good now!

**Pull request merged**